pySDC/implementations/problem_classes/AdvectionEquation_ND_FD.py

```diff
@@ -122,30 +122,26 @@ def __init__(
         self.lintol, self.liniter, self.direct_solve = lintol, liniter, direct_solver
 
         # read-only attributes
-        self._readOnly = [ndim, c, stencil_type, order]
+        self._readOnly = dict(
+            ndim=ndim, c=c, stencil_type=stencil_type, order=order)
```

Comment:
I think you need to make the dictionary keys string here

Reply:
?? That's just a simpler equivalent to

```python
self._readOnly = {'ndim': ndim, ...}
```

Reply:
Oh wait, got it